src/orange/detail/BIHData.hh

```diff
@@ -0,0 +1,108 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2021-2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/OrangeData.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/OpaqueId.hh"
+#include "corecel/Types.hh"
+#include "corecel/cont/EnumArray.hh"
+#include "corecel/data/Collection.hh"
+#include "orange/BoundingBox.hh"
+#include "orange/OrangeTypes.hh"
+
+namespace celeritas
+{
+namespace detail
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Data for a single inner node in a Bounding Interval Hierarchy.
+ */
+struct BIHInnerNode
+{
+    using real_type = float;
+
+    struct BoundingPlane
+    {
+        Axis axis;
+        real_type position;
+    };
+
+    enum class Edge
+    {
+        left,
+        right,
+        size_
+    };
+
+    BIHNodeId parent;
+
+    EnumArray<Edge, BIHNodeId> children;
+    EnumArray<Edge, BoundingPlane> bounding_planes;
```

Comment:
I wrote it this way because I thought it felt natural. A plane defined by a position and axis is seems like a pretty reasonable thing to have as a struct. I can change it two 3 arrays of independent data I suppose.

Reply:
Actually, I think that with:

```
    EnumArray<Edge, BIHNodeId> children;
    EnumArray<Edge, BoundingPlane> bounding_planes;
```

Is significantly easier to understand than:

```
    EnumArray<Edge, BIHNodeId> children;
    EnumArray<Edge, Axis> axes;
    EnumArray<Edge, real_type> positions;
```

If you think there is a real performance benefit I will change it. Otherwise I think actually having a struct describing that the positions and axes are bounding planes is preferable from a readability standpoint.